src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -51,6 +51,309 @@ public class C
+    public string Property2 { get; }
+}
+";
+            var lib = CreateCompilation(new[] { lib_cs, IsExternalInitTypeDefinition }, parseOptions: TestOptions.Regular9);
+
+            string source = @"
+public class D
+{
+    void M(C c)
+    {
+        c.Property = string.Empty;
+        c.Property2 = string.Empty;
+    }
+}
+";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular8,
+                references: new[] { useMetadataImage ? lib.EmitToImageReference() : lib.ToMetadataReference() });
+            comp.VerifyEmitDiagnostics(
+                // (6,9): error CS8852: Init-only property or indexer 'C.Property' can only be assigned in an object initializer, or on 'this' or 'base' in an instance constructor or an 'init' accessor.
+                //         c.Property = string.Empty;
+                Diagnostic(ErrorCode.ERR_AssignmentInitOnly, "c.Property").WithArguments("C.Property").WithLocation(6, 9),
+                // (7,9): error CS0200: Property or indexer 'C.Property2' cannot be assigned to -- it is read only
+                //         c.Property2 = string.Empty;
+                Diagnostic(ErrorCode.ERR_AssgReadonlyProp, "c.Property2").WithArguments("C.Property2").WithLocation(7, 9)
+                );
+        }
+
+        [Theory, CombinatorialData, WorkItem(50245, "https://github.com/dotnet/roslyn/issues/50245")]
+        public void TestCSharp8_ConsumptionInAttribute(bool useMetadataImage)
+        {
+            string lib_cs = @"
+public class TestAttribute : System.Attribute
+{
+    public int Property { get; init; }
+    public int Property2 { get; }
+}
+";
+            var lib = CreateCompilation(new[] { lib_cs, IsExternalInitTypeDefinition }, parseOptions: TestOptions.Regular9);
+
+            string source = @"
+[Test(Property = 42, Property2 = 43)]
+class C
+{
+}
+";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular8,
+                references: new[] { useMetadataImage ? lib.EmitToImageReference() : lib.ToMetadataReference() });
+            comp.VerifyEmitDiagnostics(
+                // (2,7): error CS8400: Feature 'init-only setters' is not available in C# 8.0. Please use language version 9.0 or greater.
+                // [Test(Property = 42, Property2 = 43)]
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion8, "Property = 42").WithArguments("init-only setters", "9.0").WithLocation(2, 7),
+                // (2,22): error CS0617: 'Property2' is not a valid named attribute argument. Named attribute arguments must be fields which are not readonly, static, or const, or read-write properties which are public and not static.
+                // [Test(Property = 42, Property2 = 43)]
+                Diagnostic(ErrorCode.ERR_BadNamedAttributeArgument, "Property2").WithArguments("Property2").WithLocation(2, 22)
+                );
+
+            comp = CreateCompilation(source, parseOptions: TestOptions.Regular9,
```

Comment:
>Regular9 [](start = 71, length = 8)

Consider using Latest version  instead of C# 9 for the version that supports init-only.

Reply:
Seems better to narrowly sandwich the pivotal version.

---
In reply to: [552726336](https://github.com/dotnet/roslyn/pull/50251#discussion_r552726336) [](ancestors = 552726336)